assets/src/stories-editor/blocks/amp-story-page/background-media-settings.js

```diff
@@ -121,7 +121,7 @@ const BackgroundMediaSettings = ( {
 				} )
 				.catch( () => setExtractingPoster( false ) );
 		}
-	}, [ media, mediaType, videoFeaturedImage ] );
+	}, [ media, mediaId, mediaType, mediaUrl, poster, videoFeaturedImage, isExtractingPoster, setAttributes ] );
```

Comment:
^ Here as well at least `setAttributes` is redundant, not sure about the others. Curious to know why were the other variables added here for change tracking, too. E.g. `poster`, `mediaUrl`, etc.

Reply:
These variables are all used within the effect. If they change, they change the effect's logic too.

Reply:
Yep, I can see that these are all used. Wondering if something wasn't working before due to the variables not being tracked.

Reply:
There could definitely have been times where it should have had an effect but didn't, but they might have been unreachable states due to other circumstances.

E.g. if `isExtractingPoster` was `true` and it was the only thing holding up this block `if ( media && ! media.featured_media && ! isExtractingPoster ) {`, changing that flag to `false` wouldn't have caused the effect to run again and upload the video frame. But that probably wasn't a possible state.